components/tools/OmeroPy/src/omero/plugins/metadata.py

```diff
@@ -0,0 +1,507 @@
+        return self.get_allanns(namespaces.NSMEASUREMENT)
+
+    def get_allanns(self, ns=None, anntype=None):
+        # Sooner or later we're going to end up with an object that has 1000s
+        # of annotations, and since listAnnotations() returns a generator we
+        # might as well do the same
+        for a in self.obj_wrapper.listAnnotations(ns):
+            aw = self.wrap(a)
+            if anntype and aw.get_type() != anntype:
+                continue
+            yield aw
+
+    def wrap(self, obj):
+        if obj is None:
+            return None
+        try:
+            return [self.__class__(o) for o in obj]
+        except TypeError:
+            return self.__class__(obj)
+
+    def __getattr__(self, name):
+        return getattr(self.obj_wrapper, name)
+
+    def __str__(self):
+        return "<Metadata%s>" % self.obj_wrapper
+
+
+class MetadataControl(BaseControl):
+
+    POPULATE_CONTEXTS = (
+        ("csv", populate_metadata.ParsingContext),
+        ("bulkmap", populate_metadata.BulkToMapAnnotationContext),
+        ("deletemap", populate_metadata.DeleteMapAnnotationContext),
+    )
+
+    def _configure(self, parser):
+        parser.add_login_arguments()
+
+        sub = parser.sub()
+        summary = parser.add(sub, self.summary)
+        original = parser.add(sub, self.original)
+        measures = parser.add(sub, self.measures)
+        bulkanns = parser.add(sub, self.bulkanns)
+        mapanns = parser.add(sub, self.mapanns)
+        allanns = parser.add(sub, self.allanns)
+        rois = parser.add(sub, self.rois)
+        populate = parser.add(sub, self.populate)
+        populateroi = parser.add(sub, self.populateroi)
+
+        for x in (summary, original, bulkanns, measures, mapanns, allanns,
+                  rois, populate, populateroi):
+            x.add_argument("obj",
+                           type=ProxyStringType(),
+                           help="Object in Class:ID format")
+
+        for x in (bulkanns, measures, mapanns, allanns,
+                  rois, populate, populateroi):
+            x.add_argument("--report", action="store_true", help=(
+                "Show additional information"))
+
+        for x in (rois, populate, populateroi):
+            dry_or_not = x.add_mutually_exclusive_group()
+            dry_or_not.add_argument("-n", "--dry-run", action="store_true")
+            dry_or_not.add_argument("-f", "--force", action="store_false",
+                                    dest="dry_run")
+
+        for x in (bulkanns, measures, mapanns, allanns):
+            x.add_argument(
+                "--parents", action="store_true",
+                help="Also search parents for annotations")
+
+        for x in (mapanns, allanns):
+            x.add_argument(
+                "--ns", default=None, help="Restrict to this namespace")
+            x.add_argument(
+                "--nsre",
+                help="Restrict to this namespace (regular expression)")
```

Comment:
That's an interesting feature. I wonder if it would be useful for `includeNs`, `excludeNs` in http://downloads.openmicroscopy.org/omero/5.2.0/api/slice2html/omero/cmd/graphs/ChildOption.html.